### ISSUE TITLE
Missing ARM_CONTAINER as Environment Type

### ISSUE THREAD

**Issue by kabej** [terraform-providers/terraform-provider-aws#11204]
Since 18th november, AWS Code Build support ARM.

https://aws.amazon.com/fr/about-aws/whats-new/2019/11/aws-codebuild-adds-support-for-arm--gpu--and-x-large-compute-typ/

Terraform module performs a validation and doesn't accept ARM_CONTAINER value

> expected environment.0.type to be one of [LINUX_CONTAINER LINUX_GPU_CONTAINER WINDOWS_CONTAINER], got ARM_CONTAINER

**Comment by kabej** [comment]
#11206

**Comment by bflad** [comment]
@kabej next time can you please submit feature requests following the [Feature Request issue template](https://github.com/terraform-providers/terraform-provider-aws/issues/new?labels=enhancement&template=Feature_Request.md) so contributors and maintainers have all the relevant details for searching/implementing? e.g. this affects the `aws_codebuild_project` resource and that the configuration being attempted is something like:

```hcl
resource "aws_codebuild_project" "example" {
  # ... other configuration ...

  environment {
    # ... other configuration ...
    type = "ARM_CONTAINER"
  }
}
```

Thanks so much.

**Comment by bflad** [comment]
Support for this has been merged and will release in version 2.42.0 of the Terraform AWS Provider, later this week. Thanks to @kabej for the implementation. 👍
